pkg/apiserver/webhooks.go

```diff
@@ -51,15 +54,24 @@ const (
 	serverEndpoint     = "/validate-deletion"
 )
 
-func setupWebhooks(ctx context.Context, webhookNs string, certStorageDir string) error {
-	caBytes, err := createCerts(webhookNs, certStorageDir)
-	if err != nil {
-		return err
-	}
-	if err := createValidatingWebhook(ctx, webhookNs, caBytes); err != nil {
-		return err
+var (
+	mu          sync.Mutex
+	cert        tls.Certificate
+	certModTime time.Time
+)
+
+func setupWebhooks(ctx context.Context, webhookNs string, certStorageDir string, useCertManWebhook bool) error {
+	if !useCertManWebhook {
+		caBytes, err := createCerts(webhookNs, certStorageDir)
+		if err != nil {
+			return err
+		}
+		if err := createValidatingWebhook(ctx, webhookNs, caBytes); err != nil {
```

Comment:
I think createValidatingWebhook() should still be called and the ValidatingWebhookConfiguration resource should still be created, even if you are using cert-manager for serving the webhook. The webhook should still be registered, shouldn't it?

Reply:
- as for the last point im not too sure, the validating web-hook in the "USE_CERT_MAN_FOR_WEBHOOK" operation mode will be externally defined and applied in a yaml file with the rest of the package resources. this validating webhook will require an annotation to inject the ca bundle from the cert manager certificate. without that annotation the ca bundle not be updated and the certificate will eventually expire. I have a dummy "Cert-manager porch pkg" here "https://github.com/Catalin-Stratulat-Ericsson/porch-tls" which can help a little in the explanation. as this will be a "separate" porch pkg not the default as far as i understand